karapace/backup/topic_configurations.py

```diff
@@ -0,0 +1,73 @@
+"""
+Copyright (c) 2023 Aiven Ltd
+See LICENSE for details
+"""
+from enum import Enum
+from kafka import KafkaAdminClient
+from kafka.admin import ConfigResource, ConfigResourceType
+from kafka.errors import for_code
+from kafka.protocol.admin import DescribeConfigsRequest
+from typing import Dict, Iterable, Optional
+
+
+class ConfigSource(int, Enum):
+    UNKNOWN = 0
+    TOPIC_CONFIG = 1
+    DYNAMIC_BROKER_CONFIG = 2
+    DYNAMIC_DEFAULT_BROKER_CONFIG = 3
+    STATIC_BROKER_CONFIG = 4
+    DEFAULT_CONFIG = 5
+    DYNAMIC_BROKER_LOGGER_CONFIG = 6
+
+
+DEFAULT_CONFIGS = [
+    "cleanup.policy",
+    "min.insync.replicas",
+    "retention.bytes",
+    "retention.ms",
+]
+
+
+def get_topic_configurations(
+    admin: KafkaAdminClient,
+    topic_name: str,
+    config_source_filter: Optional[Iterable[ConfigSource]] = None,
```

Comment:
```suggestion
    config_source_filter: Container[ConfigSource] = (),
```

nitpick (non-blocking): I think `Container` is slightly more appropriate here, as what we need an object that supports `in`.

nitpick (non-blocking): I think we can simplify by letting this default to the empty tuple, that way we don't need to handle `None`.